src/ucp/core/ucp_ep.c

```diff
@@ -2451,15 +2451,18 @@ int ucp_ep_config_test_rndv_support(const ucp_ep_config_t *config)
 static UCS_F_ALWAYS_INLINE int
 ucp_ep_is_am_keepalive(ucp_ep_h ucp_ep, ucp_rsc_index_t rsc_idx)
 {
+    ucp_worker_iface_t *wiface = ucp_worker_iface(ucp_ep->worker, rsc_idx);
+
     /* if we have ep2iface transport we need to send an active-message based
      * keepalive message to check the remote endpoint still exists */
-    return (ucp_ep->flags & UCP_EP_FLAG_REMOTE_ID) && /* remote ID defined */
-           (rsc_idx != UCP_NULL_RESOURCE) && /* rsc index is not NULL, i.e.
-                                                it is not a CM lane */
-           (ucp_worker_iface(ucp_ep->worker, rsc_idx)->attr.cap.flags &
-            UCT_IFACE_FLAG_CONNECT_TO_IFACE) && /* connect-to-iface */
-           (ucp_worker_iface(ucp_ep->worker, rsc_idx)->attr.cap.flags &
-            UCT_IFACE_FLAG_AM_BCOPY); /* AM Bcopy support */
+    return /* remote ID defined */
+           (ucp_ep->flags & UCP_EP_FLAG_REMOTE_ID) &&
+           /* rsc index is not NULL, i.e. it is not a CM lane */
+           (rsc_idx != UCP_NULL_RESOURCE) &&
```

Comment:
sorry, now i noticed we should check rsc_idx != UCP_NULL_RESOURCE before doing ucp_worker_iface(ucp_ep->worker, rsc_idx);

Reply:
fixed
hopefully, we will see some error with valgrind
I didn't see with just running gtest/sockaddr